### PR TITLE
Make resolver a public property

### PR DESCRIPTION
**Pull request by stevewolter**
The custom resolver mattmoor@ mentioned seems to be a very workable solution for endpoint resolving (#88). I've made a PoC implementation in http://shortn/_0SyJs3BlK7 (Google-internal link) that uses a custom resolver to resolve endpoint versions after Docker digests were resolved. Could you PTAL?

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by stevewolter** [comment]
I signed it!

File: README.md [hunk 1]
```diff
@@ -264,6 +264,37 @@ Then the references to `gcr.io/rules_k8s/server:dev` will be replaced with one
...
+  name = "dev"
+  template = ":deployment.yaml",
+  images = {
+    "gcr.io/rules_k8s/server:dev": "//server:image"
+  },
+  resolver = "//my_script",
+)
+```
+
+This script needs to invoke the default resolver (`//k8s:resolver`) with all its
```
Comment (mattmoor) [inline README.md]: I would say that this script *may* need to invoke the default resolver, if the template may also contain image references.

File: k8s/object.bzl [hunk 2]
```diff
@@ -173,7 +173,7 @@ _common_attrs = {
     # This is only needed for describe.
     "kind": attr.string(),
     "image_chroot": attr.string(),
-    "_resolver": attr.label(
+    "resolver": attr.label(
```
Comment (mattmoor) [inline k8s/object.bzl]: Can you add this to the `k8s_object` signature documentation at the bottom of the `README.md`?

Comment (stevewolter) [reply k8s/object.bzl]: Done.

File: README.md (hunk 1 above)
Comment (stevewolter) [reply README.md]: Done.

**Comment by stevewolter** [comment]
Thanks for the review! I addressed the minor comments. Let me know if there's anything else I need to be doing.

**Pull request merged**